Dockerfile.jdbc

```diff
@@ -1,4 +1,4 @@
-FROM eclipse-temurin:11 AS builder
+FROM eclipse-temurin:21-jdk-jammy AS builder
```

Comment:
1. Pin to Ubuntu Jammy so the gcc uses the same version as the Metabase image.
2. Likewise, use JDK 21 since the Metabase image uses JRE 21. This is unrelated to the glibc errors.

Reply:
2. why eclipse-temurin and not zulu-openjdk-alpine ?

Reply:
https://whichjdk.com/

Also, we can't build on alpine. DuckDB needs gcc/glibc.